### PR TITLE
[NativeAOT-LLVM] Fix a determinism issue in `lowerSpillTempsLiveAcrossSafePoints`

### PR DESCRIPTION
**Pull request by SingleAccretion**
And stop spilling frozen objects.

Encountered in #2297.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm 

Required for #2297.

File: src/coreclr/jit/llvmlower.cpp [hunk 1]
```diff
@@ -140,8 +140,22 @@ void Llvm::Lower()
...
+    struct DeterministicNodeHashInfo : public HashTableInfo<DeterministicNodeHashInfo>
+    {
+        static bool Equals(GenTree* left, GenTree* right)
+        {
+            return left == right;
+        }
+
+        static unsigned GetHashCode(GenTree* node)
+        {
+            return node->TypeGet() ^ node->OperGet();
```
Comment (yowl) [inline src/coreclr/jit/llvmlower.cpp:153]: For my benefit, how  is this value good enough to be deterministic, on the surface looks like you could have 2 nodes with the same type and operation that would not be deterministic?

Comment (SingleAccretion) [reply src/coreclr/jit/llvmlower.cpp:153]: It is deterministic in the sense that it always returns the same value for the same node.

It is ok for the iteration order to be "random" in the sense of not being related to insertion order, what was not ok is the it run-to-run variance introduced by the lower bits of node pointers being different.

**Review by yowl — APPROVED** [review]
Thanks

**Pull request merged**